lib/rules/number-leading-zero/index.js

```diff
@@ -77,13 +91,28 @@ const rule = function(expectation) {
         }
 
         if (expectation === "never") {
-          const match = /(?:\D|^)(0+\.\d+)/.exec(valueNode.value);
+          const regexp = /((?:\D|^))(0+)(\.\d+)/;
+          const match = regexp.exec(valueNode.value);
 
           if (match === null) {
             return;
           }
 
-          const capturingGroupIndex = match[0].length - match[1].length;
+          if (context.fix) {
+            if (node.value) {
+              node.value = node.value.replace(regexp, "$1$3");
+              return;
+            }
+
+            if (node.params) {
```

Comment:
From coverage reports, it's never true:
https://coveralls.io/builds/13380484/source?filename=lib%2Frules%2Fnumber-leading-zero%2Findex.js#L107